### PR TITLE
chore: validate onBlur and only conditionally onChange

### PR DESCRIPTION
**Pull request by hein-j**
*Description of changes:*
- Validate on blur
- Don't validate onChange unless there is error

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cwoolum — APPROVED** [review]
LGTM

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts [hunk 1]
```diff
@@ -421,6 +422,57 @@ export const addFormAttributes = (component: StudioComponent | StudioComponentCh
   return attributes;
 };
 
+/**
+  if (errors.name?.hasError) {
+    await runValidationTasks("name", value);
+  }
+ */
```
Comment (lyonsbp) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:429]: Was this comment left intentionaly?

**Review by lyonsbp — APPROVED** [review]

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts (hunk 1 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:429]: Yes, those kind of comments are basically representations of what the ast writer will output

**Review by lyonsbp — APPROVED** [review]

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 2]
```diff
@@ -101,9 +104,12 @@ export default function CustomDataForm(props) {
           defaultValue=\\"johndoe@amplify.com\\"
           onChange={async (e) => {
             const { value } = e.target;
-            await runValidationTasks(\\"email\\", value);
+            if (errors.email?.hasError) {
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:107]: making sure I'm interpreting this correctly, onChange will only run validations if there is currently an error. If it briefly becomes valid, onChange validation will stop happening and will instead only validate once more during the onBlur event.

Comment (hein-j) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:107]: Yes, that is correct

**Review by awinberg-aws — APPROVED** [review]

**Review by Jshhhh — APPROVED** [review]

**Pull request merged**